src/pepr/operator/controllers/keycloak/authservice/authorization-policy.ts

```diff
@@ -0,0 +1,164 @@
+  name: string,
+  namespace: string,
+): AuthorizationPolicy {
+  return {
+    kind: "AuthorizationPolicy",
+    metadata: {
+      name: `${name}-authservice`,
+      namespace,
+    },
+    spec: {
+      action: Action.Custom,
+      provider: {
+        name: "authservice",
+      },
+      rules: [
+        {
+          when: [
+            {
+              key: "request.headers[authorization]",
+              notValues: ["*"],
+            },
+          ],
+        },
+      ],
+      selector: {
+        matchLabels: labelSelector,
+      },
+    },
+  };
+}
+
+function jwtAuthZAuthorizationPolicy(
+  labelSelector: { [key: string]: string },
+  name: string,
+  namespace: string,
+): AuthorizationPolicy {
+  return {
+    kind: "AuthorizationPolicy",
+    metadata: {
+      name: `${name}-jwt-authz`,
+      namespace,
+    },
+    spec: {
+      selector: {
+        matchLabels: labelSelector,
+      },
+      rules: [
+        {
+          from: [
+            {
+              source: {
+                requestPrincipals: [`https://sso.${UDSConfig.domain}/realms/uds/*`],
+              },
+            },
+          ],
+        },
+      ],
+    },
+  };
+}
+
+function authNRequestAuthentication(
+  labelSelector: { [key: string]: string },
+  name: string,
+  namespace: string,
+): RequestAuthentication {
+  return {
+    kind: "RequestAuthentication",
+    metadata: {
+      name: `${name}-jwt-authn`,
+      namespace,
+    },
+    spec: {
+      jwtRules: [
+        {
+          forwardOriginalToken: true,
+          issuer: `https://sso.${UDSConfig.domain}/realms/uds`,
+          jwksUri: `https://sso.${UDSConfig.domain}/realms/uds/protocol/openid-connect/certs`,
+        },
+      ],
+      selector: {
+        matchLabels: labelSelector,
+      },
+    },
+  };
+}
+
+async function updatePolicy(
+  event: AuthServiceEvent,
+  labelSelector: { [key: string]: string },
+  pkg: UDSPackage,
+) {
+  const operation = event.action === AuthServiceAction.Add ? "Apply" : "Delete";
+  const namespace = pkg.metadata!.namespace!;
+  const generation = (pkg.metadata?.generation ?? 0).toString();
+  const ownerReferences = getOwnerRef(pkg);
+  const getKind = (kind: string | undefined) => {
+    return kind && kind === "RequestAuthentication" ? RequestAuthentication : AuthorizationPolicy;
+  };
+
+  try {
+    [
+      authserviceAuthorizationPolicy(labelSelector, event.name, namespace),
+      authNRequestAuthentication(labelSelector, event.name, namespace),
+      jwtAuthZAuthorizationPolicy(labelSelector, event.name, namespace),
+    ].forEach(async p => {
+      p!.metadata!.ownerReferences = ownerReferences;
+      p!.metadata!.labels = {
+        "uds/package": pkg.metadata!.name!,
+        "uds/generation": generation,
+      };
+      const kind = getKind(p.kind);
+      await K8s(kind)[operation](p);
+    });
+  } catch (e) {
+    Log.error(e, `Failed to update auth policy for ${event.name} in ${namespace}: ${e}`);
+  }
```

Comment:
I think this is clearer. `getKind()` isn't really typesafe if there are more policy types added in the future, this avoids that by removing the loop and switching what the helper function is doing.

If you want to make long lines shorter you can add a temp variable, but I don't really have strong opinion on that part.

```suggestion

  function updateMetadata(resource: RequestAuthentication | AuthorizationPolicy) {
    resource!.metadata!.ownerReferences = getOwnerRef(pkg);
    resource!.metadata!.labels = {
      "uds/package": pkg.metadata!.name!,
      "uds/generation": generation,
    };
    return resource;
  }

  try {
    await K8s(AuthorizationPolicy)[operation](updateMetadata(authserviceAuthorizationPolicy(labelSelector, event.name, namespace)));
    await K8s(RequestAuthentication)[operation](updateMetadata(authNRequestAuthentication(labelSelector, event.name, namespace)));
    await K8s(AuthorizationPolicy)[operation](updateMetadata(jwtAuthZAuthorizationPolicy(labelSelector, event.name, namespace)));
  } catch (e) {
    Log.error(e, `Failed to update auth policy for ${event.name} in ${namespace}: ${e}`);
  }
```

Reply:
Updated.  Definitely looks better.